store/layered_store_hints.go

```diff
@@ -9,3 +9,23 @@ const (
 	LSH_NO_CACHE LayeredStoreHint = iota
 	LSH_MASTER_ONLY
 )
+
+func hintsContains(hints []LayeredStoreHint, contains LayeredStoreHint) bool {
+	for _, hint := range hints {
+		if hint == contains {
+			return true
+		}
+	}
+	return false
+}
+
+func hintsContainsAny(hints []LayeredStoreHint, contains ...LayeredStoreHint) bool {
```

Comment:
You mean using sets? The side of both of these arrays is always going to be low. So that would be super overkill.